### ISSUE TITLE
JavaScript test runners skip resolution validations

### ISSUE THREAD

**Issue by enzocanoo** [Microsoft/Recognizers-Text#607]
From [`function getParserTestRunner(extractor, parser)`](https://github.com/Microsoft/Recognizers-Text/blob/master/JavaScript/test/runner-datetime.js#L103) in `runner-datetime.js`

```javascript
...
if (actual.value && expected.Value) {
	// timex
	t.is(actual.value.timex, expected.Value.Timex, 'Result.Value.Timex');

	// resolutions
	var actualValue = {
		timex: actual.value.timex,
		futureResolution: actual.value.futureResolution,
		pastResolution: actual.value.pastResolution,
	}

	t.deepEqual(actualValue.futureResolution, expected.Value.FutureResolution);
	t.deepEqual(actualValue.pastResolution, expected.Value.PastResolution);
}
...
```

If the tested parser does not resolve the correct values for a test case and `actual.value` is null, then the other assertions are not executed, causing a false positive.

The same behavior is presented in [`getMergedParserTestRunner`](https://github.com/Microsoft/Recognizers-Text/blob/master/JavaScript/test/runner-datetime.js#L140) and [`getModelTestRunner`](https://github.com/Microsoft/Recognizers-Text/blob/master/JavaScript/test/runner-datetime.js#L170)

The correct behavior would be that those tests failed because `actual.value` is different from `expected.Value`

**Comment by amitstein** [comment]
There is another issue in Merged Parse test. 
The test is checking _TypeName_ property https://github.com/Microsoft/Recognizers-Text/blob/06abe6ad30f29bee5f28119d370da6e73150c2d8/JavaScript/test/runner-datetime.js#L101 when the correct name is _type_.
https://github.com/Microsoft/Recognizers-Text/blob/06abe6ad30f29bee5f28119d370da6e73150c2d8/JavaScript/packages/recognizers-date-time/src/dateTime/baseMerged.ts#L342
It is a false positive since the property TypeName is _undefined_ for both objects.

**Comment by enzocanoo** [comment]
Fixed with:
- Fix Number-with-Unit tests assertions #628
- [JS] Fix datetime asserts #646

The same issue was fixed in Python with the following PR:
- Fix number with unit test assertions #629
- Fix datetime asserts #643
